packages/smart-wallet/src/typeGuards.js

```diff
@@ -0,0 +1,65 @@
+// @ts-check
+
+import { M } from '@agoric/store';
+import {
+  InstanceHandleShape,
+  ProposalShape,
+} from '@agoric/zoe/src/typeGuards.js';
+
+export const shape = {
+  // smartWallet
+  StringCapData: {
+    body: M.string(),
+    slots: M.arrayOf(M.string()),
+  },
+
+  // invitations
+  ContractInvitationSpec: M.split(
```

Comment:
Separately it seems that we have "Spec" as another genericish thing in the space of class/record/interface/shape/whatever. That ought to drop out of our naming or get clear guidance of when to use it in types. (post launch?)

Reply:
the distinction here is meaningful. the "Spec" is a specification for producing the thing and not the thing itself. it comes up in the wallet code because we send messages to describe things to do.

e.g. `ContractInvitationSpec` says that to get the invitation, use the `instance` and call `publicInvitationMaker` with `invitationArgs` arguments. 

I'm open to changing the name but  I don't see it as a blocker for PS0 and definitely not this PR about implementing argument validation. If you like I can make a ticket to audit the smart-wallet package coding style.